src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs

```diff
@@ -0,0 +1,422 @@
+                        _fileTraceWriter.Verbose(t.Exception.ToString());
+                        return;
+                    }
+
+                    _fileTraceWriter.Verbose("Packages restored.");    
+                    _reloadScript();
+                });
+        }
+
+        public override async Task Invoke(object[] parameters)
+        {
+            object input = parameters[0];
+            TraceWriter traceWriter = (TraceWriter)parameters[1];
+            IBinder binder = (IBinder)parameters[2];
+            ExecutionContext functionExecutionContext = (ExecutionContext)parameters[3];
+
+            try
+            {
+                _fileTraceWriter.Verbose("Function started");
+
+                // if there are any binding parameters in the output bindings,
+                // parse the input as json to get the binding data
+                var bindingData = new Dictionary<string, string>();
+                if (_outputBindings.Any(p => p.HasBindingParameters) ||
+                    _inputBindings.Any(p => p.HasBindingParameters))
+                {
+                    bindingData = GetBindingData(input);
+                }
+                bindingData["InvocationId"] = functionExecutionContext.InvocationId.ToString();
+
+                MethodInfo function = GetFunctionTarget();
+
+                object[] arguments = await BindArgumentsAsync(binder, function, bindingData, parameters);
+
+                object functionResult = function.Invoke(null, arguments);
+
+                if (functionResult is Task)
+                {
+                    functionResult = await ((Task)functionResult)
+                        .ContinueWith(t => GetTaskResult(t));
+                }
+
+                var functionOutputs = functionResult as IDictionary<string, object>;
+
+                // If we have a single output binding, directly specify its value
+                if (functionResult != null && _outputBindings.Count == 1)
+                {
+                    var outputBinding = _outputBindings.Single();
+
+                    if (functionOutputs == null || !functionOutputs.ContainsKey(outputBinding.Name))
+                    {
+                        functionOutputs = new Dictionary<string, object>
+                        {
+                            {outputBinding.Name, functionResult }
+                        };
+                    }
+                }
+                else if (functionOutputs == null)
+                {
+                    var output = function.GetParameters().Where(p => p.IsOut).ToList();
+                    functionOutputs = output.ToDictionary(o => o.Name, o => arguments[o.Position]);
+                }
+
+                await ProcessOutputBindingsAsync(_outputBindings, input, binder, bindingData, functionOutputs);
+
+                _fileTraceWriter.Verbose("Function completed (Success)");
+            }
+            catch (CompilationErrorException ex)
+            {
```

Comment:
do you think this compilation exception block would make more sense inside of the GetFunctionTarget method (or wherever this is being compiled?)